### PR TITLE
Feature/arc42 documentation

### PR DESCRIPTION
**Pull request by bs-sili**

### CODE REVIEW

File: docs/1-introduction-and-goals.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+## 1 Introduction and goals
+
+The Digital Twins are no longer administrated in a central registry
+anymore. The implementation of two services, "Discovery Finder" and "BPN
```
Comment (LuLeRoemer) [inline docs/1-introduction-and-goals.md:4]: "no longer" and "anymore" express the same meaning, right? :)

File: docs/1-introduction-and-goals.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    (horizontal and vertical scalability included) and meant to help to
+    connect companies. So, it is integrated into the Catena-X network.
+
+### Stakeholders
+
+| Role             | Description                                              | Goal, Intention                                      |
+|------------------|----------------------------------------------------------|------------------------------------------------------|
+| Consumer         | uses a BPN Discovery                                     | wants to find a BPN                                  |
+| Data Provider    | Registers his keys of a specific type in a BPN Discovery | wants that his EDC / Digital Twin can be found       |
+| Catena-X network |                                                          | to enable the companies to connect and exchange data |
```
Comment (LuLeRoemer) [inline docs/1-introduction-and-goals.md:34]: No description?

File: docs/2-architecture-constraints.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
+## 2 Architecture and constraints
+
+### Overall Architecture
+
+A consumer sends a request to the Discovery Finder with a catalogue of a certain types e.g., "bpid", "oen". The Discovery Finder
```
Comment (LuLeRoemer) [inline docs/2-architecture-constraints.md:5]: type

File: docs/2-architecture-constraints.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+responses with a list of endpoints of BPN Discoveries which provide BPNs for these kind of types.  
+Subsequently the consumer sends a request to the given BPN Discoveries. If the BPN Discovery finds a matching entries he returns a list of BPNs.
```
Comment (LuLeRoemer) [inline docs/2-architecture-constraints.md:7]: entry

File: docs/2-architecture-constraints.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+-   Must be compliant and fulfill the Catena-X Guidelines.
+
+-   The deployment is done in a Catena-X environment. So, the BPN
+    Discovery must be able to run in a Kubernetes environment with Helm
+    Charts.
+
+-   The BPN Discovery must be managed centralized.
+
+-   Keycloak as an OAuth2 compliant authorization is needed to manage the identity of and access of the user.
```
Comment (LuLeRoemer) [inline docs/2-architecture-constraints.md:50]: ... manage the identity ~~of~~ and access of the user.

File: docs/5-deployment-view.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+-   Webserver
+
+-   Kubernetes-Cluster
+
+-   Helm
+
+To deploy this system, you need to use the Helm Chart in a running
+Kubernetes cluster. The Helm Chart is located under "charts/bpndiscovery".
+In case you don't have a running cluster, you can set up one by yourself
+locally, using minikube. For further information checkout the readme.md.
```
Comment (LuLeRoemer) [inline docs/5-deployment-view.md:14]: Maybe set a direct link to the README.md here using markdown :)

File: docs/6-concepts.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+### Overall Concept
+
+#The overall concept can be found under **2 Architecture and
+constraints**.
+
+### BPN Discovery API
+
+The BPN Discovery is a central component in the environment. Every
+consumer and data provider can call ths service via the given api
```
Comment (LuLeRoemer) [inline docs/6-concepts.md:11]: this

File: docs/6-concepts.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+endpoints as single entry.
+
+This service is allowed to **administrators only**.
+
+Path variable {resourceId} is required
+
+Response: http Status Code 204 -> no response
+
+An OpenApi.yaml for this service can be found under
+"backend/src/main/resources/static/bpn-discovery-service-openapi.yaml"
```
Comment (LuLeRoemer) [inline docs/6-concepts.md:167]: Maybe set a direct link here.

File: docs/6-concepts.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+
+### Security / Safety / Use the JWT Token to get the BPN from requester
+
+The whole environment is secured with Keycloak. The BPN Discovery has a
```
Comment (LuLeRoemer) [inline docs/6-concepts.md:171]: Let's rather write OAuth2 here, and that we recommend using Keycloak. Keycloak just implements the OAuth2 standard.

File: docs/6-concepts.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+
+1. [ ] *VIEW*( "view_bpn_discovery" )
+2. [ ] *ADD*( "add_bpn_discovery" )
+3. [ ] *DELETE*( "delete_bpn_discovery" )
+
+These roles and the BPN of the requester are send within a JWT Token
+generated by Keycloak.
+
+With all three roles it is possible to add and delete entries from the
+BPN Discovery Database. To ensure only the own entries can be deleted
```
Comment (LuLeRoemer) [inline docs/6-concepts.md:185]: ~~To ensure only the own entries can be deleted~~ To ensure that a user can only delete twins that he created, the BPN of the database entry and the BPN from the Token are compared.

File: docs/documentation.md [hunk 11]
```diff
@@ -2,4 +2,81 @@
...
+The service is secured by a OAuth2 compliant authorization. Every API calls has to provide a
+valid Bearer Token. Authorization is provided by a role based access. These roles are possible:
+
+| Role                      | Description                            |
+|---------------------------|----------------------------------------|
+| view_discovery_endpoint   | can search for BPN Discovery endpoints |
+| add_discovery_endpoint    | can add BPN Discovery endpoints        |
+| delete_discovery_endpoint | can delete BPN Discovery endpoints     |
+
+The Bearer Token also provide a claim with the BPN. This BPN is uses to store entries.
```
Comment (LuLeRoemer) [inline docs/documentation.md:77]: used

File: docs/documentation.md [hunk 12]
```diff
@@ -2,4 +2,81 @@
...
+valid Bearer Token. Authorization is provided by a role based access. These roles are possible:
+
+| Role                      | Description                            |
+|---------------------------|----------------------------------------|
+| view_discovery_endpoint   | can search for BPN Discovery endpoints |
+| add_discovery_endpoint    | can add BPN Discovery endpoints        |
+| delete_discovery_endpoint | can delete BPN Discovery endpoints     |
+
+The Bearer Token also provide a claim with the BPN. This BPN is uses to store entries. 
+When an entries shall be deleted this BPN is used to verify that the requester is allowed to do so.
```
Comment (LuLeRoemer) [inline docs/documentation.md:78]: entry

File: docs/documentation.md [hunk 13]
```diff
@@ -2,4 +2,81 @@
...
+
+## Deployment
+To deploy this system, you need to use the Helm Chart in a running
+Kubernetes cluster. The Helm Chart is located under "charts/bpndiscovery". For further information checkout the readme.md.
```
Comment (LuLeRoemer) [inline docs/documentation.md:82]: Again, maybe link the README.md

**Review by LuLeRoemer — CHANGES_REQUESTED** [review]
Accroding to TRG 7.07 we should include a license header in the MD files: https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-07
Maybe let's discuss this.

File: docs/6-concepts.md (hunk 10 above)
Comment (bs-sili) [reply docs/6-concepts.md:185]: hm, wouldn't call it "twins". Cause in this service only BPNs are saved to a specific type / key - combination like "oen" / "oen-23" -> "BPN1234"

**Review by LuLeRoemer — APPROVED** [review]
LGTM

**Review by tunacicek — APPROVED** [review]
LGTM

**Pull request merged**